Controller/ThemeController.php

```diff
@@ -40,18 +41,32 @@ class ThemeController
      */
     protected $cookieOptions;
 
+    /**
+     * @var RouterInterface
+     */
+    protected $router;
+
+    /**
+     * @var string|null
+     */
+    protected $defaultRoute;
+
     /**
      * Theme controller construct.
      *
      * @param ActiveTheme $activeTheme   active theme instance
      * @param array       $themes        Available themes
      * @param array|null  $cookieOptions The options of the cookie we look for the theme to set
+     * @param RouterInterface $router
+     * @param string|null $defaultRoute
      */
-    public function __construct(ActiveTheme $activeTheme, array $themes, array $cookieOptions = null)
+    public function __construct(ActiveTheme $activeTheme, array $themes, array $cookieOptions = null, RouterInterface $router, $defaultRoute = null)
```

Comment:
it looks like a BC break, shall we announce it somehow, or refactor ?

Reply:
or allow the argument to be `null` and in that case fall back to the old logic